### PR TITLE
add rest pagination snippet

### PR DESCRIPTION
**Pull request by ddebrunner**
Adds snippet showing pagination with `@rest`.

Initially just `PAGE_NUMBER`, will add others once this is merged (as others will be almost carbon copies so want to get first approved).

### CODE REVIEW

File: rest/pagination/api.graphql [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+`first` (number of nodes to fetch) and `after` (opaque cursor indicating
+the starting point).
+"""
+type CustomerConnection {
+  edges: [CustomerEdge]
+  pageInfo: PageInfo!
+}
+
+"""
+`CustomerEdge` provides access to the node and its cursor.
```
Comment (bobbiejc) [inline rest/pagination/api.graphql:35]: Stopped review here, but wanted to get what I have so far to you

**Review by bobbiejc — COMMENTED** [review]
Didn't finish,but looks good, some minor word changes, etc, take/leave as you like

File: rest/pagination/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+The easiest way to use paginated fields is with a GraphQL operation
+that has variables for the `first` and `after` arguments. Examples
+are show in `requests.graphql`.
+
+For the first call (to get the first page) the operation is called
+with a value of `first` to set the number of nodes in a page
+and either an empty string (`""`) for `after` or the `after` variable is omitted.
+
+Then to fetch subsequent pages keep the value of `first` fixed and set
+`after` to the value of `endCursor` from `pageInfo` from the previouse request.
```
[the remark this answers is not in the document]
Comment (ddebrunner) [reply rest/pagination/README.md:77]: I want to keep the "keeping it fixed" as the recommended approach.

File: rest/pagination/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Query fields that return a connection type must have two field arguments, `first` for
+the number of nodes to fetch and `after` an opaque cursor. With StepZen these
+arguments are available in the context of the `@rest` configuration with the values
+converted to the correct value for the REST API.
+
+For example with `PAGE_NUMBER` the `after` opaque cursor value the client uses
+is converted to a integer page number in the context of `@rest`. So if the
+REST API expects URL query arguments `pageSize` and `pageNumber` then `@rest` would have
+an `endpoint` such as:
```
[the remark this answers is not in the document]
Comment (ddebrunner) [reply rest/pagination/README.md:30]: Not sure about the wording, the client actually sends the opaque cursor (which it has also received), but the first part tends to imply that the client might have to convert it.

File: rest/pagination/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+      "name":"Fred"
+    },
+    {
+      "name":"Barney"
+    }
+  ]
+}
+```
+
+The field `Query.customersPageNumber` is an example of `PAGE_NUMBER` pagination.
```
[the remark this answers is not in the document]
Comment (ddebrunner) [reply rest/pagination/README.md:56]: That's covered in the field definitions.

[the remark this answers is not in the document]
Comment (ddebrunner) [reply rest/pagination/README.md:56]: The dot notation is the standard proposal
https://github.com/graphql/graphql-wg/blob/main/rfcs/SchemaCoordinates.md

File: rest/pagination/api.graphql (hunk 1 above)
Comment (ddebrunner) [reply rest/pagination/api.graphql:35]: Thanks!

File: rest/pagination/api.graphql [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+# @rest supports three styles of pagination for REST responses:
+#
+# PAGE_NUMBER - pages by requesting a specific page number
+# OFFSET - pages by reqesting a starting record offset
+# NEXT_CURSOR - pages by opaque cursors returned by the REST API
+#
+# In each case @rest allows the backend REST API to be exposed
+# in the GraphQL schema as a field using
+# GraphQL Cursor Connections Specification.
+# https://relay.dev/graphql/connections.htm
```
[the remark this answers is not in the document]
Comment (ddebrunner) [reply rest/pagination/api.graphql:12]: Changed

File: rest/pagination/README.md [hunk 6]
```diff
@@ -67,15 +67,12 @@ TODO
...
 
 The easiest way to use paginated fields is with a GraphQL operation
-that has variables for the `first` and `after` arguments. Examples
+that has variables for the arguments `first` and `after`. Examples
 are show in `requests.graphql`.
 
-For the first call (to get the first page) the operation is called
-with a value of `first` to set the number of nodes in a page
-and either an empty string (`""`) for `after` or the `after` variable is omitted.
+To retrieve the first page of a result, the operation is called with `after` omitted or set to the empty string (`""`), and `first`, and for all calls, is set to the number of results to return in a page.
```
Comment (bobbiejc) [inline rest/pagination/README.md:73]:
```suggestion
To retrieve the first page of a result, the operation is called with `after` omitted or set to the empty string (`""`), and for all calls, `first` is set to the number of results to return.
```

**Review by bobbiejc — COMMENTED** [review]
one minor nit, then GTG

File: rest/pagination/README.md (hunk 6 above)
Comment (ddebrunner) [reply rest/pagination/README.md]: to be clearer, separated out first for subsequent requests.

File: rest/pagination/README.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
+# Pagination with `@rest`
+
+This snippet shows how `@rest` fields are configured to support pagination.
```
Comment (bobbiejc) [inline rest/pagination/README.md:3]: Is there a better description than `@rest` field?   Can we provide a link to our documentation or a snippet for @rest?

"how fields defined with StepZen's @rest directive" maybe?

File: rest/pagination/README.md [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+## GraphQL pagination
+
+StepZen supports pagination through the standard
+[GraphQL Cursor Connection Specification](https://relay.dev/graphql/connections.htm).
+
+This allows clients consuming a StepZen GraphQL endpoint to page through results
```
Comment (bobbiejc) [inline rest/pagination/README.md:10]: Can this sentence merge with previous sentence in same paragraph?

**Review by bobbiejc — APPROVED** [review]
I just added some suggestions, but no need to hold this up for them.

**Pull request merged**